### DISCUSSION TITLE
Got memory leak warning for SafeAccessTokenHandle

### DISCUSSION THREAD

**Discussion by ChristianHoe** [dotnet/runtime#42116] [General]
Why leaves the following code a memory footprint for each call? Tested with .net 472 and core 3.1
            
```
var handle = new SafeAccessTokenHandle(IntPtr.Zero);
handle.Dispose();
```

**Comment by ericstj** [comment]
`SafeAccessTokenHandle` is a class, so each call to new will allocate an object on the managed heap.  Those objects won't be cleaned up until GC runs and executes the finalizer for the SafeHandle class.

If you run:
```
GC.Collect();
GC.WaitForPendingFinalizers();
```
Do you observe the memory consumption go back down?  cc @Maoni0

**Reply by ChristianHoe** [reply]
Nope. I added the GC lines. Same result for SafeAccessTokenHandle .

I would expect the Diagnostic Tools -> Memory Usage -> Take Snapshot - Feature doing the same in the background. Am I wrong?

**Comment by ChristianHoe** [comment]
Same problem for:

```
var registry = Registry.CurrentUser.OpenSubKey("Software");
registry.Dispose();
```

**Reply by ericstj** [reply]
I suspect it's a problem with how you're measuring this.  You can create a commandline application which does this in a loop and observe that usage is constant.  

```c#
using System;
using Microsoft.Win32;

namespace handleTest
{
    class Program
    {
        static void Main(string[] args)
        {
            while(true)
            {
                var registry = Registry.CurrentUser.OpenSubKey("Software");
                registry.Dispose();
            }
        }
    }
}
```

When I do this I see the application's private bytes stay at 8300 KB.  Tested on .NETCore 3.1.

**Reply by ChristianHoe** [reply]
Learned something new. In the context of (static) main you always end with an extra instance for each created variable... Thanks for your patience. And I have to go to step one.
